### PR TITLE
Liberating pipeline

### PR DESCRIPTION
**Pull request by wilko77**
We had two problems with PR from external repos:
1) they were not allowed to push to codecov.
2) they were not allowed to publish to the artifact feed on Azure.

I changed the pipeline to:
1) allow publishing to codecov to fail. (This should not block)
2) we only publish to the feed from the master branch. (This makes sense as we only want to publish artifacts that have been reviewed.)

### CODE REVIEW

File: .azurePipeline/unittest_wheel_steps.yml [hunk 1]
```diff
@@ -35,8 +35,9 @@ steps:
       opSysFlag=$(echo ${{ parameters.operatingSystem }} | sed 's/[[:punct:]]//g' | tr '[:upper:]' '[:lower:]' | head -c30)
       pyVFlag=$(echo python${{ parameters.pythonVersion }} | sed 's/[[:punct:]]//g' | tr '[:upper:]' '[:lower:]' | head -c30)
       echo "codecov flags: $opSysFlag,$pyVFlag"
-      python -m codecov --token $(CODECOV_TOKEN) \
-        --file coverage.xml \
+      bash <(curl -s https://codecov.io/bash) \
```
Comment (JoyceXinyueWang) [inline .azurePipeline/unittest_wheel_steps.yml:38]: no need for token anymore?

File: azurePipeline.yml [hunk 2]
```diff
@@ -1,12 +1,3 @@
-# This pipeline should be triggered by every local branches, but should not be used for external PRs.
-# In fact, this pipeline is able to use some of our secrets to push images on dockerhub and deploy on k8s.
-pr: none
-trigger:
-  branches:
-    include:
-    - '*'
```
Comment (JoyceXinyueWang) [inline azurePipeline.yml:7]: why do you delete this section? The pipeline will no longer be triggered by PRs?

**Review by JoyceXinyueWang — APPROVED** [review]
It seems Azure and Travis are happy with this so I will approve it

File: .azurePipeline/unittest_wheel_steps.yml (hunk 1 above)
Comment (wilko77) [reply .azurePipeline/unittest_wheel_steps.yml:38]: the codecov tool reads the token from the environment variable. So no need to specify it here. -> Less clutter. :)

File: azurePipeline.yml (hunk 2 above)
Comment (wilko77) [reply azurePipeline.yml:7]: It's actually the opposite. The `pr: none` bit prevented the pipeline to be triggered by a PR.  However, we want every PR to trigger the tests. External PRs won't have access to the secrets, thus this is fine. 
By removing all of this, we get the default behavior, which is run for every PR.

**Pull request merged**